src/main/scala/com/github/unchama/seichiassist/subsystems/vote/subsystems/fairy/bukkit/actions/BukkitRecoveryMana.scala

```diff
@@ -18,223 +16,219 @@ import com.github.unchama.seichiassist.subsystems.vote.subsystems.fairy.domain.p
+    val isManaExistsSeventyFivePercent = oldManaAmount.ratioToCap.exists(_ >= 0.75)
+
+    val defaultAmount = Math.pow(playerLevel.level / 10, 2)
+
+    val chainVoteDayNumber = chainVoteNumber.value
+    // 連続投票を適用した除算量
+    val chainVoteDivisionAmount =
+      if (chainVoteDayNumber >= 30) 2
+      else if (chainVoteDayNumber >= 10) 1.5
+      else if (chainVoteDayNumber >= 3) 1.25
+      else 1
+
+    // りんごの開放状況を適用した除算量
+    val appleOpenStateDivisionAmount =
+      if (isAppleOpenStateIsOpenOrOpenALittle && isManaExistsSeventyFivePercent) 2
+      else 1
+
+    // りんごの開放状況まで適用したりんごの消費量 (暫定)
+    val appleOpenStateReflectedAmount =
+      (defaultAmount / chainVoteDivisionAmount).toInt / appleOpenStateDivisionAmount
+
+    // 妖精がつまみ食いする量
+    val amountEatenByKnob =
+      if (appleOpenStateReflectedAmount >= 10)
+        new Random().nextInt(appleOpenStateReflectedAmount / 10)
+      else 0
+
+    /*
+       最終的に算出されたりんごの消費量
+        (現時点では持っているりんごの数を
+        考慮していないので消費量は確定していない)
+     */
+    Math.max(appleOpenStateReflectedAmount + amountEatenByKnob, 1)
+  }
+
+  /**
+   * MineStackに入っているがちゃりんごの数を考慮した
+   * がちゃりんごの消費量を計算します。
+   */
+  private def computeFinallyAppleConsumptionAmount(appleConsumptionAmount: Int): F[Int] = for {
+    appleOpenState <- fairyAPI.appleOpenState(uuid)
+    gachaRingoObject <- LiftIO[F].liftIO {
+      MineStackObjectList.findByName("gachaimo")
     }
+  } yield {
+    val mineStackedGachaRingoAmount =
+      playerdata.minestack.getStackedAmountOf(gachaRingoObject.get)
+
+    // りんごの消費量
+    if (appleOpenState == AppleOpenState.NotOpen) {
+      0
+    } else {
+      if (mineStackedGachaRingoAmount > appleConsumptionAmount)
+        appleConsumptionAmount
+      else mineStackedGachaRingoAmount
+    }
+  }
+
+  /**
+   * マナの回復量を計算します。
+   */
+  private def computeManaRecoveryAmount(appleConsumptionAmount: Int): F[Int] = for {
+    defaultRecoveryManaAmount <- fairyAPI.fairyRecoveryMana(uuid)
+    appleOpenState <- fairyAPI.appleOpenState(uuid)
+    oldManaAmount <- ContextCoercion {
+      manaApi.readManaAmount(player)
+    }
+    gachaRingoObject <- LiftIO[F].liftIO {
+      MineStackObjectList.findByName("gachaimo")
+    }
+  } yield {
+    val isAppleOpenStateIsOpenOrOpenALittle =
+      appleOpenState == AppleOpenState.OpenALittle || appleOpenState == AppleOpenState.Open
+    val isManaExistsSeventyFivePercent = oldManaAmount.ratioToCap.exists(_ >= 0.75)
+
+    val mineStackedGachaRingoAmount =
+      playerdata.minestack.getStackedAmountOf(gachaRingoObject.get)
+
+    // マナの回復量を算出する
+
+    val appleOpenStateDivision = {
+      if (isAppleOpenStateIsOpenOrOpenALittle && isManaExistsSeventyFivePercent) 2
+      else if (appleOpenState == AppleOpenState.NotOpen) 4
+      else 1
+    }
+
+    val reflectedAppleOpenStateAmount =
+      defaultRecoveryManaAmount.recoveryMana / appleOpenStateDivision
+
+    // minestackに入っているりんごの数を適用したマナの回復量
+    val reflectedMineStackedAmount =
```

Comment:
```suggestion
    val mineStackBasedRegenValue =
```

nit: `MineStacked`は疑問の余地がある一般的ではない命名です